core/runtime/src/main/java/io/quarkus/runtime/configuration/Services.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.runtime.configuration;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class Services {
+    private static final Map<String, Service> services = new ConcurrentHashMap<>();
+
+    private Services() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static Optional<Service> get(String name) {
```

Comment:
The registration may happen at any time, and there is not way for a consumer to _listen_ for the registration.